### DISCUSSION TITLE
Display default widgets as button

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#2820] [Q&A] [answered]
Hi there,
Is there a way to move Tree-View and Properties widgets to a button just like there is camera button...
I'm using web viewer to create application. i want few widgets to be placed in the default place but few to be displayed as buttons which on clicking would display the  contained tree or properties data as it now does in widget .

![image](https://user-images.githubusercontent.com/27068008/144437793-716f76d5-abca-45dc-8d65-6ef10004f7e9.png)

**Comment by FlyersPh9** [comment, marked as answer]
That's an interesting proposal, we'll take this into consideration.

**Reply by kmurari103** [reply]
I see Design Review has something like that, we also want to achieve this using itwin-web-viewer. When clicked it should show the data, on clicking again it should get minimized. 
![image](https://user-images.githubusercontent.com/27068008/144452163-e9db940d-b082-410c-ae64-8323c8336877.png)

**Reply by aruniverse** [reply]
Design Review is using an older version of our UI design system. You can enable it in the viewer but it should be noted this design system is being deprecated and you will be unable to enable it in the next major version of the viewer. If you want to enable it there's a prop on the viewer called 'uiFrameworkVersion' set that to "1". You'll notice it looks kinda funky with the built in widgets we provide, so you'll have to hide them and re add them yourself.

**Comment by kmurari103** [comment]
I assume its not supported yet.
So is there a way around like popup-panel or something where i can attach these tree view & properties tabs which would get popped up when that button is clicked??

**Reply by bsteinbk** [reply]
There is a way in UI 2.0, also known as AppUi, to define a button with custom content. See these [examples](https://github.com/iTwin/itwinjs-core/blob/f60482bfb6fb100c054020c9372d01d5c1a78541/test-apps/ui-test-app/src/frontend/appui/frontstages/ViewsFrontstage.tsx#L322) in ui-test-app. We have no built-in way to automatically support putting a widget in that popup. You could look at the following [example component](https://github.com/iTwin/itwinjs-core/blob/c20d12c24994f96e57c6ba1d3400f6c6bf674773/test-apps/ui-test-app/src/frontend/appui/widgets/VisibilityWidget.tsx#L44) as a starting point for building your own popup button.
